### DISCUSSION TITLE
Oracle connection

### DISCUSSION THREAD

**Discussion by liwen-wang** [sfu-db/connector-x#371] [Q&A] [answered]
Hi,

I have used your package to successfully connect my oracle database, and get table data much faster than pandas. But when I connect the second oracle database, I met a trouble and got 
RuntimeError: parse error: invalid port number

The first conn str looks like;
url='xxx.site.com:1521/devel'
conn=f'oracle://{user}:{password}@{url}'

The second conn str I used is a custom jdbc url:
url='(DESCRIPTION=(LOAD_BALANCE=on)(FAILOVER=ON)(ADDRESS=(PROTOCOL=TCP)(HOST=abc-ef.xyz.com)(PORT=1521))(CONNECT_DATA=(SERVER=DEDICATED)(SERVICE_NAME=mm.xyz.com)))'
conn=f'oracle://{user}:{password}@{url}'

Both connections strings worked fine on pandas.

How would I update the above connection string to make it work?
Thank you for your help

Liwen

**Comment by wangxiaoying** [comment, marked as answer]
Hi @liwen-wang , I think the first connection is correct. Is there any special characters in your connection string? Can you try to url encode each part of the `conn`?

**Comment by liwen-wang** [comment]
Hi Xiaoying,

You are right. After encoding each part of connection, it works now.

Thank you very much.
Liwen

From: Xiaoying Wang ***@***.***>
Date: Thursday, October 13, 2022 at 12:47 PM
To: sfu-db/connector-x ***@***.***>
Cc: Wang, Liwen ***@***.***>, Mention ***@***.***>
Subject: Re: [sfu-db/connector-x] Oracle connection (Discussion #371)
***CAUTION:*** This email is not from a BCM Source. Only click links or open attachments you know are safe.
________________________________

Hi @liwen-wang<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_liwen-2Dwang&d=DwMCaQ&c=ZQs-KZ8oxEw0p81sqgiaRA&r=t6o15idjqfhNZ0-bBhFHPA&m=IpcMLeY5brE1wYHDYqEt1m2LBSvzTnfCXOwCmMbfRGjmkibDiv_TMtV0fe08027H&s=a02a-H4uOO6gUSKjSYKyeT4D6uVqWTB6MTy67ABxwn4&e=> , I think the first connection is correct. Is there any special characters in your connection string? Can you try to url encode each part of the conn?
